### PR TITLE
alphabetization and minor refactoring of the Features section

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change re-orders the Features section of the documentation nav-bar to help with discovery of articles.

This will not effect the 2.0.0 or 2.1.0 documentation sub-sites.

### CODE REVIEW

File: Documentation/toc.yml [hunk 1]
```diff
@@ -176,8 +154,40 @@
       items:
       - name: Controller Mapping tool
         href: Tools/ControllerMappingTool.md 
-  - name: Scene Transition Service
-    href: Extensions/SceneTransitionService/SceneTransitionServiceOverview.md
+  - name: Utility
+    items:
+    - name: Detecting Platform Capabilities
+      href: DetectingPlatformCapabilities.md
```
Comment (wiwei) [inline Documentation/toc.yml]: It feels like "Utility" is a bit too general and isn't descriptive enough for what's actually contained there - if someone wanted to find detecting platform capabilities, I don't see how they would know to drill into a somewhat vaguely named "Utilities" dropdown.

Also in general, I know that this change will folder-ize things that were top-level before in order to create some consistency with other folder things. I feel like in general we should aim to get users to the thing in the fewest number of clicks possible (or rather, when there are two solutions, one of which involves fewer clicks, we should favor those). Given this, it feels like maybe we should keep those current top-level non folder things as non-folder things (i.e. Teleport System only has an overview, no need to have it expand and require another click to get to what's interesting).

The alphabetical order makes a ton of sense and I think think creates clarity on expectations (for both readers and contributors).

Comment (davidkline-ms) [reply Documentation/toc.yml]: i can make that change. thanks!

Comment (davidkline-ms) [reply Documentation/toc.yml]: One note on single entry folders... I am going to keep the Extension services folder as there are a couple of new ones in 2.2.0 that need to be documented

Comment (davidkline-ms) [reply Documentation/toc.yml]: as part of the extension folder, i am going to duplicate the extension tool wizard to help elevate it's visibility

**Pull request merged**